lib/memo_wise.rb

```diff
@@ -78,6 +78,22 @@ def initialize(#{all_args})
     end
   HEREDOC
 
+  module CreateMemoWiseStateOnExtended
+    def extended(base)
+      MemoWise::InternalAPI.create_memo_wise_state!(base)
+      super
+    end
+  end
+  private_constant(:CreateMemoWiseStateOnExtended)
+
+  module CreateMemoWiseStateOnInherited
+    def inherited(subclass)
+      super
```

Comment:
I know this is how we defined it in the `module_eval`, but I'm wondering if there's a reason to keep the `super` before the `create_memo_wise_state!` here but after it above. If either works, I don't care which we stick with but think it would be nice to be consistent.